openssl/src/x509/mod.rs

```diff
@@ -1069,6 +1069,28 @@ impl X509ReqRef {
             X509NameRef::from_ptr(name)
         }
     }
+
+    /// Returns the public key of the certificate request.
+    ///
+    /// This corresponds to [`X509_REQ_get_pubkey"]
+    ///
+    /// [`X509_REQ_get_pubkey`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_REQ_get_pubkey.html
+    pub fn public_key(&self) -> PKey<Public> {
+        unsafe {
+            let key = ffi::X509_REQ_get_pubkey(self.as_ptr());
+            PKey::from_ptr(key)
+        }
+    }
+
+    /// Returns the extensions of the certificate request.
+    ///
+    /// This corresponds to [`X509_REQ_get_extensions"]
+    pub fn extensions(&self) -> Stack<X509Extension> {
```

Comment:
Oh right, this looks like it needs to specifically return `Result<Stack<X509Extension>, ErrorStack>`.